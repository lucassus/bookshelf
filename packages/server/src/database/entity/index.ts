export { Author } from "./Author";
export { Avatar } from "./Avatar";
export { Book } from "./Book";
export { BookCopy } from "./BookCopy";
export { User } from "./User";
