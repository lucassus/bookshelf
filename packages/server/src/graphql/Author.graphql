type Author {
  id: ID!
  name: String!
  bio: String
  photo: Image
  books: [Book!]!
}
