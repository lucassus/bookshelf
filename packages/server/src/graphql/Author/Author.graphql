type Author implements Resource {
  id: ID!
  name: String!
  bio: String
  photo: Image
  books: [Book!]!
}
