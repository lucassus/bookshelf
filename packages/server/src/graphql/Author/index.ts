export { resolvers } from "./Author.resolvers";
