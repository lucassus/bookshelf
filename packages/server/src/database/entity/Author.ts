import {
  Column,
  CreateDateColumn,
  Entity,
  OneToMany,
  PrimaryGeneratedColumn,
  UpdateDateColumn
} from "typeorm";

import { Book } from "./Book";

@Entity({ name: "authors" })
export class Author {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column({ default: "" })
  bio: string;

  @Column({ name: "photo_path" })
  photoPath: string;

  @OneToMany(() => Book, (book) => book.author)
  books: Promise<Book[]>;

  @CreateDateColumn({ name: "created_at" })
  created_at: Date;

  @UpdateDateColumn({ name: "updated_at" })
  updated_at: Date;
}
