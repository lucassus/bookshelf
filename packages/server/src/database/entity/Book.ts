import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn
} from "typeorm";

import { Author } from "./Author";

@Entity({ name: "books", orderBy: { title: "ASC" } })
export class Book {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  title: string;

  @Column({ name: "cover_path" })
  coverPath: string;

  @ManyToOne(() => Author, (author) => author.books)
  @JoinColumn({ name: "author_id" })
  author: Promise<Author>;

  @Column({ name: "author_id" })
  authorId: number;

  @Column({ default: false })
  favourite: boolean;

  @CreateDateColumn({ name: "created_at" })
  created_at: Date;

  @UpdateDateColumn({ name: "updated_at" })
  updated_at: Date;
}
