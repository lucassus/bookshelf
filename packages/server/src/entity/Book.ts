import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn
} from "typeorm";

import { Author } from "./Author";

@Entity({ name: "books" })
export class Book {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  title: string;

  @Column({ name: "cover_path" })
  coverPath: string;

  @Column({ name: "author_id" })
  authorId: number;

  @ManyToOne(() => Author, (author) => author.books)
  @JoinColumn({ name: "author_id" })
  author: Promise<Author>;

  @CreateDateColumn({ name: "created_at" })
  created_at: Date;

  @UpdateDateColumn({ name: "updated_at" })
  updated_at: Date;
}
