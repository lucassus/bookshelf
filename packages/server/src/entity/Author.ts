import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from "typeorm";

import { Book } from "./Book";

@Entity()
export class Author {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column({ name: "photo_path" })
  photoPath: string;

  @OneToMany(() => Book, (book) => book.author)
  books: Book[];
}
