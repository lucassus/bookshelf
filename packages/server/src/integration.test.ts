import { ApolloServer, gql } from "apollo-server-express";
import { createTestClient } from "apollo-server-testing";
import { getConnection } from "typeorm";

import { Author } from "./database/entity/Author";
import { Book } from "./database/entity/Book";
import { createServer } from "./server";

let server: ApolloServer;

beforeEach(async () => {
  const connection = getConnection();
  server = createServer(connection);
});

it("fetches books", async () => {
  // Given
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        booksCount
        books {
          id
          title
          cover {
            url
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data).toMatchSnapshot();
});

it("fetches a book", async () => {
  // Given
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        book(id: 2) {
          id
          title
          description
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data).toMatchSnapshot();
});

it("responds with error when book cannot be found", async () => {
  // Given
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        book(id: 200) {
          id
          title
        }
      }
    `
  });

  // Then
  expect(res.data).toBe(null);
  expect(res.errors![0].message).toEqual(
    'Could not find any entity of type "Book" matching: 200'
  );
});

it("fetches authors along with books", async () => {
  // Given
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        authors {
          name
          books {
            title
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data).toMatchSnapshot();
});

it("fetches an author", async () => {
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        author(id: 1) {
          id
          name
          bio
          books {
            title
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data).toMatchSnapshot();
});

it("fetches books along with authors and books again", async () => {
  // Given
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        books {
          title
          author {
            name
            books {
              title
            }
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data).toMatchSnapshot();
});

it("fetches a random book", async () => {
  // Given
  const connection = getConnection();
  await connection.createQueryBuilder().delete().from(Book).execute();
  const author = await connection.manager.findOneOrFail(Author, {
    name: "Andrzej Sapkowski"
  });
  await connection.manager.save(Book, {
    authorId: author.id,
    title: "The tower of the swallow",
    coverPath: "/images/book-covers/witcher4.jpg"
  });

  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        randomBook {
          title
          cover {
            url
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data!.randomBook).toMatchSnapshot();
});

it("fetches users", async () => {
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        users {
          name
          email
          info
          avatar {
            color
            image {
              url
            }
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data!.users).toMatchSnapshot();
});

it("fetches a user", async () => {
  const { query } = createTestClient(server);

  // When
  const res = await query({
    query: gql`
      query {
        user(id: 1) {
          name
          email
          info
          avatar {
            color
            image {
              url
            }
          }
        }
      }
    `
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data!.user).toMatchSnapshot();
});

it("updates book favourite", async () => {
  // Given
  const connection = getConnection();

  const book = await connection.manager.findOneOrFail(Book, 1);
  expect(book.favourite).toBe(false);

  const { mutate } = createTestClient(server);

  // When
  const res = await mutate({
    mutation: gql`
      mutation {
        updateBookFavourite(id: ${book.id}, favourite: true) {
          id
          title
          favourite
        }
      }
    `,
    variables: { id: book.id, favourite: true }
  });

  // Then
  expect(res.data).not.toBeUndefined();
  expect(res.data!.updateBookFavourite).toMatchSnapshot();

  const updatedBook = await connection.manager.findOneOrFail(Book, 1);
  expect(updatedBook.favourite).toBe(true);
});
