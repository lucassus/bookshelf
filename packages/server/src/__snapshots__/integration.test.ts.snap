// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches a book 1`] = `
Object {
  "book": Object {
    "description": "The Dursleys were so mean and hideous that summer that all Harry Potter wanted was to get back to the Hogwarts School for Witchcraft and Wizardry. But just as he's packing his bags, Harry receives a warning from a strange, impish creature named Dobby who says that if Harry Potter returns to Hogwarts, disaster will strike

      And strike it does. For in Harry's second year at Hogwarts, fresh torments and horrors arise, including an outrageously stuck-up new professor, Gilderoy Lockhart, a spirit named Moaning Myrtle who haunts the girls' bathroom, and the unwanted attentions of Ron Weasley's younger sister, Ginny.
      
      But each of these seem minor annoyances when the real trouble begins, and someone -- or something -- starts turning Hogwarts students to stone. Could it be Draco Malfoy, a more poisonous rival than ever? Could it possibly be Hagrid, whose mysterious past is finally told? Or could it be the one everyone at Hogwarts most suspects . . . Harry Potter himself?",
    "id": 2,
    "title": "Harry Potter and the Chamber of Secrets",
  },
}
`;

exports[`fetches a random book 1`] = `
Object {
  "cover": Object {
    "url": "http://examples.devmastery.pl/assets/images/book-covers/witcher4.jpg",
  },
  "title": "The tower of the swallow",
}
`;

exports[`fetches a user 1`] = `
Object {
  "avatar": Object {
    "color": "yellow",
    "image": Object {
      "url": "http://examples.devmastery.pl/assets/images/avatars/w13.png",
    },
  },
  "email": "alice@example.com",
  "info": "Food scholar. Incurable tv fanatic. Reader. Typical zombie buff. Gamer. Lifelong creator. Certified organizer.",
  "name": "Alice",
}
`;

exports[`fetches an author 1`] = `
Object {
  "author": Object {
    "bio": "Although she writes under the pen name J.K. Rowling, pronounced like rolling, her name when her first Harry Potter book was published was simply Joanne Rowling. Anticipating that the target audience of young boys might not want to read a book written by a woman, her publishers demanded that she use two initials, rather than her full name. As she had no middle name, she chose K as the second initial of her pen name, from her paternal grandmother Kathleen Ada Bulgen Rowling. She calls herself Jo and has said, \\"No one ever called me 'Joanne' when I was young, unless they were angry.\\" Following her marriage, she has sometimes used the name Joanne Murray when conducting personal business. During the Leveson Inquiry she gave evidence under the name of Joanne Kathleen Rowling. In a 2012 interview, Rowling noted that she no longer cared that people pronounced her name incorrectly.

      Rowling was born to Peter James Rowling, a Rolls-Royce aircraft engineer, and Anne Rowling (née Volant), on 31 July 1965 in Yate, Gloucestershire, England, 10 miles (16 km) northeast of Bristol. Her mother Anne was half-French and half-Scottish. Her parents first met on a train departing from King's Cross Station bound for Arbroath in 1964. They married on 14 March 1965. Her mother's maternal grandfather, Dugald Campbell, was born in Lamlash on the Isle of Arran. Her mother's paternal grandfather, Louis Volant, was awarded the Croix de Guerre for exceptional bravery in defending the village of Courcelles-le-Comte during the First World War.
      
      Rowling's sister Dianne was born at their home when Rowling was 23 months old. The family moved to the nearby village Winterbourne when Rowling was four. She attended St Michael's Primary School, a school founded by abolitionist William Wilberforce and education reformer Hannah More. Her headmaster at St Michael's, Alfred Dunn, has been suggested as the inspiration for the Harry Potter headmaster Albus Dumbledore.
      
      As a child, Rowling often wrote fantasy stories, which she would usually then read to her sister. She recalls that: \\"I can still remember me telling her a story in which she fell down a rabbit hole and was fed strawberries by the rabbit family inside it. Certainly the first story I ever wrote down (when I was five or six) was about a rabbit called Rabbit. He got the measles and was visited by his friends, including a giant bee called Miss Bee.\\" At the age of nine, Rowling moved to Church Cottage in the Gloucestershire village of Tutshill, close to Chepstow, Wales. When she was a young teenager, her great aunt, who Rowling said \\"taught classics and approved of a thirst for knowledge, even of a questionable kind,\\" gave her a very old copy of Jessica Mitford's autobiography, Hons and Rebels. Mitford became Rowling's heroine, and Rowling subsequently read all of her books.
      
      Rowling has said of her teenage years, in an interview with The New Yorker, \\"I wasn’t particularly happy. I think it’s a dreadful time of life.\\" She had a difficult homelife; her mother was ill and she had a difficult relationship with her father (she is no longer on speaking terms with him). She attended secondary school at Wyedean School and College, where her mother had worked as a technician in the science department. Rowling said of her adolescence, \\"Hermione [a bookish, know-it-all Harry Potter character] is loosely based on me. She's a caricature of me when I was eleven, which I'm not particularly proud of.\\" Steve Eddy, who taught Rowling English when she first arrived, remembers her as \\"not exceptional\\" but \\"one of a group of girls who were bright, and quite good at English.\\" Sean Harris, her best friend in the Upper Sixth owned a turquoise Ford Anglia, which she says inspired the one in her books.",
    "books": Array [
      Object {
        "title": "Harry Potter and the Chamber of Secrets",
      },
      Object {
        "title": "Harry Potter and the Deathly Hallows",
      },
      Object {
        "title": "Harry Potter and the Goblet of Fire",
      },
      Object {
        "title": "Harry Potter and the Half-Blood Prince",
      },
      Object {
        "title": "Harry Potter and the Order of the Phoenix",
      },
      Object {
        "title": "Harry Potter and the Prisoner of Azkaban",
      },
      Object {
        "title": "Harry Potter and the Sorcerer's Stone",
      },
    ],
    "id": 1,
    "name": "J. K. Rowling",
  },
}
`;

exports[`fetches authors along with books 1`] = `
Object {
  "authors": Array [
    Object {
      "books": Array [
        Object {
          "title": "Harry Potter and the Chamber of Secrets",
        },
        Object {
          "title": "Harry Potter and the Deathly Hallows",
        },
        Object {
          "title": "Harry Potter and the Goblet of Fire",
        },
        Object {
          "title": "Harry Potter and the Half-Blood Prince",
        },
        Object {
          "title": "Harry Potter and the Order of the Phoenix",
        },
        Object {
          "title": "Harry Potter and the Prisoner of Azkaban",
        },
        Object {
          "title": "Harry Potter and the Sorcerer's Stone",
        },
      ],
      "name": "J. K. Rowling",
    },
    Object {
      "books": Array [
        Object {
          "title": "Abaddon's Gate",
        },
        Object {
          "title": "Babylon's Ashes",
        },
        Object {
          "title": "Caliban's War",
        },
        Object {
          "title": "Cibola Burn",
        },
        Object {
          "title": "Leviathan Wakes",
        },
        Object {
          "title": "Nemesis Games",
        },
        Object {
          "title": "Persepolis Rising",
        },
        Object {
          "title": "Tiamat's Wrath",
        },
      ],
      "name": "James S. A. Corey",
    },
    Object {
      "books": Array [
        Object {
          "title": "Baptism of fire",
        },
        Object {
          "title": "Blood of Elves",
        },
        Object {
          "title": "The lady of the lake",
        },
        Object {
          "title": "The tower of the swallow",
        },
        Object {
          "title": "Time of contempt",
        },
      ],
      "name": "Andrzej Sapkowski",
    },
  ],
}
`;

exports[`fetches books 1`] = `
Object {
  "books": Array [
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/expanse3.jpg",
      },
      "id": 10,
      "title": "Abaddon's Gate",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/expanse6.jpg",
      },
      "id": 13,
      "title": "Babylon's Ashes",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/witcher3.jpg",
      },
      "id": 18,
      "title": "Baptism of fire",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/witcher1.jpg",
      },
      "id": 16,
      "title": "Blood of Elves",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/expanse2.jpg",
      },
      "id": 9,
      "title": "Caliban's War",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/expanse4.jpg",
      },
      "id": 11,
      "title": "Cibola Burn",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/harry2.jpg",
      },
      "id": 2,
      "title": "Harry Potter and the Chamber of Secrets",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/harry7.jpg",
      },
      "id": 7,
      "title": "Harry Potter and the Deathly Hallows",
    },
    Object {
      "cover": Object {
        "url": "http://examples.devmastery.pl/assets/images/book-covers/harry4.jpg",
      },
      "id": 4,
      "title": "Harry Potter and the Goblet of Fire",
    },
  ],
  "booksCount": 20,
}
`;

exports[`fetches books along with authors and books again 1`] = `
Object {
  "books": Array [
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Abaddon's Gate",
          },
          Object {
            "title": "Babylon's Ashes",
          },
          Object {
            "title": "Caliban's War",
          },
          Object {
            "title": "Cibola Burn",
          },
          Object {
            "title": "Leviathan Wakes",
          },
          Object {
            "title": "Nemesis Games",
          },
          Object {
            "title": "Persepolis Rising",
          },
          Object {
            "title": "Tiamat's Wrath",
          },
        ],
        "name": "James S. A. Corey",
      },
      "title": "Abaddon's Gate",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Abaddon's Gate",
          },
          Object {
            "title": "Babylon's Ashes",
          },
          Object {
            "title": "Caliban's War",
          },
          Object {
            "title": "Cibola Burn",
          },
          Object {
            "title": "Leviathan Wakes",
          },
          Object {
            "title": "Nemesis Games",
          },
          Object {
            "title": "Persepolis Rising",
          },
          Object {
            "title": "Tiamat's Wrath",
          },
        ],
        "name": "James S. A. Corey",
      },
      "title": "Babylon's Ashes",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Baptism of fire",
          },
          Object {
            "title": "Blood of Elves",
          },
          Object {
            "title": "The lady of the lake",
          },
          Object {
            "title": "The tower of the swallow",
          },
          Object {
            "title": "Time of contempt",
          },
        ],
        "name": "Andrzej Sapkowski",
      },
      "title": "Baptism of fire",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Baptism of fire",
          },
          Object {
            "title": "Blood of Elves",
          },
          Object {
            "title": "The lady of the lake",
          },
          Object {
            "title": "The tower of the swallow",
          },
          Object {
            "title": "Time of contempt",
          },
        ],
        "name": "Andrzej Sapkowski",
      },
      "title": "Blood of Elves",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Abaddon's Gate",
          },
          Object {
            "title": "Babylon's Ashes",
          },
          Object {
            "title": "Caliban's War",
          },
          Object {
            "title": "Cibola Burn",
          },
          Object {
            "title": "Leviathan Wakes",
          },
          Object {
            "title": "Nemesis Games",
          },
          Object {
            "title": "Persepolis Rising",
          },
          Object {
            "title": "Tiamat's Wrath",
          },
        ],
        "name": "James S. A. Corey",
      },
      "title": "Caliban's War",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Abaddon's Gate",
          },
          Object {
            "title": "Babylon's Ashes",
          },
          Object {
            "title": "Caliban's War",
          },
          Object {
            "title": "Cibola Burn",
          },
          Object {
            "title": "Leviathan Wakes",
          },
          Object {
            "title": "Nemesis Games",
          },
          Object {
            "title": "Persepolis Rising",
          },
          Object {
            "title": "Tiamat's Wrath",
          },
        ],
        "name": "James S. A. Corey",
      },
      "title": "Cibola Burn",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Harry Potter and the Chamber of Secrets",
          },
          Object {
            "title": "Harry Potter and the Deathly Hallows",
          },
          Object {
            "title": "Harry Potter and the Goblet of Fire",
          },
          Object {
            "title": "Harry Potter and the Half-Blood Prince",
          },
          Object {
            "title": "Harry Potter and the Order of the Phoenix",
          },
          Object {
            "title": "Harry Potter and the Prisoner of Azkaban",
          },
          Object {
            "title": "Harry Potter and the Sorcerer's Stone",
          },
        ],
        "name": "J. K. Rowling",
      },
      "title": "Harry Potter and the Chamber of Secrets",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Harry Potter and the Chamber of Secrets",
          },
          Object {
            "title": "Harry Potter and the Deathly Hallows",
          },
          Object {
            "title": "Harry Potter and the Goblet of Fire",
          },
          Object {
            "title": "Harry Potter and the Half-Blood Prince",
          },
          Object {
            "title": "Harry Potter and the Order of the Phoenix",
          },
          Object {
            "title": "Harry Potter and the Prisoner of Azkaban",
          },
          Object {
            "title": "Harry Potter and the Sorcerer's Stone",
          },
        ],
        "name": "J. K. Rowling",
      },
      "title": "Harry Potter and the Deathly Hallows",
    },
    Object {
      "author": Object {
        "books": Array [
          Object {
            "title": "Harry Potter and the Chamber of Secrets",
          },
          Object {
            "title": "Harry Potter and the Deathly Hallows",
          },
          Object {
            "title": "Harry Potter and the Goblet of Fire",
          },
          Object {
            "title": "Harry Potter and the Half-Blood Prince",
          },
          Object {
            "title": "Harry Potter and the Order of the Phoenix",
          },
          Object {
            "title": "Harry Potter and the Prisoner of Azkaban",
          },
          Object {
            "title": "Harry Potter and the Sorcerer's Stone",
          },
        ],
        "name": "J. K. Rowling",
      },
      "title": "Harry Potter and the Goblet of Fire",
    },
  ],
}
`;

exports[`fetches users 1`] = `
Array [
  Object {
    "avatar": Object {
      "color": "yellow",
      "image": Object {
        "url": "http://examples.devmastery.pl/assets/images/avatars/w13.png",
      },
    },
    "email": "alice@example.com",
    "info": "Food scholar. Incurable tv fanatic. Reader. Typical zombie buff. Gamer. Lifelong creator. Certified organizer.",
    "name": "Alice",
  },
  Object {
    "avatar": Object {
      "color": "green",
      "image": Object {
        "url": "http://examples.devmastery.pl/assets/images/avatars/m10.png",
      },
    },
    "email": "bob@example.com",
    "info": "Twitter fan. Social media expert. Hardcore explorer. Communicator. Amateur coffee lover.",
    "name": "Bob",
  },
  Object {
    "avatar": Object {
      "color": "red",
      "image": Object {
        "url": "http://examples.devmastery.pl/assets/images/avatars/w2.png",
      },
    },
    "email": "celine@example.com",
    "info": "Wannabe pop culture specialist. Tv scholar. Lifelong social media practitioner. Music geek. Falls down a lot.",
    "name": "Celine",
  },
  Object {
    "avatar": Object {
      "color": "blue",
      "image": Object {
        "url": "http://examples.devmastery.pl/assets/images/avatars/m25.png",
      },
    },
    "email": "dan@example.com",
    "info": "Professional problem solver. Devoted explorer. Gamer. Unapologetic bacon guru.",
    "name": "Dan",
  },
]
`;

exports[`updates book favourite 1`] = `
Object {
  "favourite": true,
  "id": 1,
  "title": "Harry Potter and the Sorcerer's Stone",
}
`;
